standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreChecker.java

```diff
@@ -468,6 +459,25 @@ void findUnknownPartitions(Table table, Set<Path> partPaths, byte[] filterExp,
         }
       }
     }
+    
+    Set<Path> partPathsInMS = partPaths;
+    // don't want the table dir
+    partPathsInMS.remove(tablePath);
+    // remove partition paths in partPathsInMS, to getPartitionsNotOnFs
+    partPathsInMS.removeAll(allPartDirs);
+    // There can be edge case where user can define partition directory outside of table directory
+    // to avoid eviction of such partitions
+    // we check for partition path not exists and add to result for getPartitionsNotOnFs.
+    for (Path partPath : partPathsInMS) {
+      FileSystem fs = partPath.getFileSystem(conf);
```

Comment:
As you pointed out, Having multi-threaded check of partition files will be useful.